src/components/Navbar/Navbar.jsx

```diff
@@ -1,17 +1,55 @@
+import { useRef, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import PropTypes from "prop-types";
 import { Input } from "@components/ui/form";
+import { searchTask } from "@features/filter/filterSlice";
+import { TASK_SEARCH_DELAY_IN_MS } from "@helpers/constants";
+import purify from "@helpers/text/purify";
+import useDebounce from "@hooks/useDebounce";
 import "./Navbar.scss";
 
-function Navbar() {
+function Navbar({ onSearching }) {
     const PLACEHOLDER = "Search...";
+    const dispatch = useDispatch();
+    const textToSearch = useSelector((state) => state.filter.search);
+    const [text, setText] = useState(textToSearch);
+    const typingTimerRef = useRef();
```

Comment:
why it's required?

Reply:
I've added it to show a loader while searching for tasks. I've seen it on the [Figma board](https://www.figma.com/file/9Rwb2k9Dfbh8Ycb6kbJfzZ/Todo-App-UI-for-New-Joiner?type=design&node-id=0-1&mode=design.).
So, the idea here is that the loader will be shown until the user has stopped writing. As there isn't any API call, I had to delay it in this way. Thank you!

[Screencast from 10-31-2023 11:49:02 AM.webm](https://github.com/sse-wtag/todo-app-react/assets/144016889/71fd831f-3875-450b-8e55-719062150ece)